### PR TITLE
Improve messages.inc and message_stocks.inc documentation

### PR DESCRIPTION
**Pull request by OciXCrom**
Right now these two files have minimal documentation for just a few natives inside them. I added the full documentation that's needed for them.

### CODE REVIEW

File: plugins/include/messages.inc [hunk 1]
```diff
@@ -18,88 +18,584 @@
...
-/* Lets you directly hook a message in the engine!
- * You can overwrite the message before anything happens and either let the message continue
- * or fully block it. Here is how it works:
- * If you hook a message, the message is stored but not sent. You have the opportunity to
- * not only execute code, but to get/set the contents of the message, before you choose to 
- * either block it or let it go on its way. The hooked function will be passed a msg_id, msg_dest, and entity index. 
- * The return value can be passed to unregister_message() to stop the message from being hooked */
+/**
+ * Lets you directly hook a message in the engine.
+ * You can overwrite the message before anything happens and either let the
```
Comment (Arkshine) [inline plugins/include/messages.inc]: Should be a  @note

File: plugins/include/messages.inc [hunk 2]
```diff
@@ -18,88 +18,584 @@
...
- * There are eight different ways of sending a message, five are ints, two are floats, and one is string.
- * These are denoted by iArgType.  argn is the number
- * of the argument. Exceeding the bounds of 1 to get_msg_args() is a bad idea. 
- * As of AMX Mod X 1.5, the middle parameter of set_* no longer does anything.
- * You cannot change the message argument type (as this would crash the mod anyway)
+/**
+ * Gets number of arguments that were passed to a message.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc (hunk 1 above)
Comment (Arkshine) [reply plugins/include/messages.inc]: Also, the callback info should be more obvious and descriptive like https://github.com/alliedmodders/amxmodx/blob/master/plugins/include/cellarray.inc#L441-L473

File: plugins/include/messages.inc [hunk 3]
```diff
@@ -18,88 +18,584 @@
...
+ *                       success, or 0 if an invalid message id is passed
+ * @error                If the specified function can't be found, an
+ *                       error will be thrown.
+ */
 native register_message(iMsgId, const szFunction[]);
 
-/* Unregisters a message hook previously created with register_message
- * You must pass the proper message id, and return value from the message to unregister the message successfully. */
+/**
+ * Unregisters a message hook previosly created with register_message().
```
Comment (Arkshine) [inline plugins/include/messages.inc]: previously

File: plugins/include/messages.inc [hunk 4]
```diff
@@ -18,88 +18,584 @@
...
-
-/* Gets number of arguments that were passed to this message */
 native get_msg_args();
 
-/* Gets the argument type of argument argn */
+/**
+ * Gets the argument type of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 5]
```diff
@@ -18,88 +18,584 @@
...
+ * @return               Argument type (see ARG_* constants in message_const.inc)
+ */
 native get_msg_argtype(argn);
 
-/* Gets the value of argn. */
+/**
+ * Gets the integer value of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 6]
```diff
@@ -18,88 +18,584 @@
...
+ * @error                If an invalid message argument is passed, an
+ *                       error will be thrown.
+ */
 native get_msg_arg_int(argn);
+
+/**
+ * Gets the float value of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 7]
```diff
@@ -18,88 +18,584 @@
...
+ * @error                If an invalid message argument is passed, an
+ *                       error will be thrown.
+ */
 native Float:get_msg_arg_float(argn);
+
+/**
+ * Gets the string value from a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 8]
```diff
@@ -18,88 +18,584 @@
...
+ *                       error will be thrown.
+ */
 native get_msg_arg_string(argn, szReturn[], iLength);
 
-/* sets the value of argn. */
+/**
+ * Sets the integer value of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 9]
```diff
@@ -18,88 +18,584 @@
...
+ * @error                If an invalid message argument is passed, an
+ *                       error will be thrown.
+ */
 native set_msg_arg_int(argn, argtype, iValue);
+
+/**
+ * Sets the float value of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 10]
```diff
@@ -18,88 +18,584 @@
...
+ * @error                If an invalid message argument is passed, an
+ *                       error will be thrown.
+ */
 native set_msg_arg_float(argn, argtype, Float:fValue);
+
+/**
+ * Sets the string value of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

File: plugins/include/messages.inc [hunk 11]
```diff
@@ -18,88 +18,584 @@
...
+ *                       error will be thrown.
+ */
 native set_msg_arg_string(argn, const szString[]);
 
-/* Gets the origin of a message */
+/**
+ * Gets the origin of a message.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       tt should never be used unless inside a registered message function.
```
Comment (Arkshine) [inline plugins/include/messages.inc]: it

**Comment by OciXCrom** [comment]
@Arkshine All done!

File: plugins/include/messages.inc [hunk 12]
```diff
@@ -18,88 +18,588 @@
...
 native set_msg_arg_float(argn, argtype, Float:fValue);
+
+/**
+ * Sets the string value of a specified argument.
+ *
+ * @note This function will fail if used outside a hooked message scope, thus
+ *       it should never be used unless inside a registered message function.
+ *
+ * @param argn           Argument number
+ * @param argtype        Argument type (see ARG_* constants in message_const.inc)
```
Comment (Arkshine) [inline plugins/include/messages.inc]: 2 arguments only here.

Comment (OciXCrom) [reply plugins/include/messages.inc]: @Arkshine Fixed.

File: plugins/include/message_stocks.inc [hunk 13]
```diff
@@ -16,7 +16,15 @@
 #endif
 #define _message_stocks_included
 
-/* Creates a death message. */
+/**
+ * Creates a death message.
```
Comment (WPMGPRoSToTeMa) [inline plugins/include/message_stocks.inc]: Maybe sends instead of creates?

File: plugins/include/messages.inc [hunk 14]
```diff
@@ -18,88 +18,587 @@
...
+ *
+ * @note The function is called in the following manner:
+ *   msg_id          - Message id
+ *   msg_dest        - Destination type (see MSG_* constants in messages_const.inc)
+ *   msg_entity      - Entity receiving the message
+ *
+ * @note You can overwrite the message before anything happens by using the
+ *       set_msg_arg_* functions and either let the message continue by
+ *       returning PLUGIN_CONTINUE or fully block it with PLUGIN_HANDLED.
+ * @note If you hook a message, the message is stored but not sent. You have
```
Comment (WPMGPRoSToTeMa) [inline plugins/include/messages.inc]: Maybe we should add a reference about `register_event` here? Something like: "It is the opposite to register_event. register_event executes your code after the message is sent."

**Comment by Arkshine** [comment]
Looks enough good now, thanks.

**Pull request merged**